libraries/ts-command-line/src/test/__snapshots__/CommandLineParameter.test.ts.snap

```diff
@@ -73,90 +155,181 @@ Object {
+Object {
+  "argumentName": "NUMBER",
+  "defaultValue": 123,
+  "description": "An integer with a default",
+  "environmentVariable": "ENV_INTEGER",
+  "kind": 2,
+  "longName": "--integer-with-default",
+  "required": false,
+  "shortName": undefined,
+  "value": 123,
+  "values": undefined,
+}
+`;
+
+exports[`CommandLineParameter parses an input with NO parameters 7`] = `
+Object {
+  "argumentName": "NUMBER",
+  "defaultValue": undefined,
+  "description": "An integer",
+  "environmentVariable": undefined,
+  "kind": 2,
+  "longName": "--integer-required",
+  "required": true,
+  "shortName": undefined,
+  "value": 123,
+  "values": undefined,
+}
+`;
+
+exports[`CommandLineParameter parses an input with NO parameters 8`] = `
 Object {
   "argumentName": "TEXT",
+  "defaultValue": undefined,
   "description": "A string",
+  "environmentVariable": "ENV_INTEGER",
   "kind": 3,
   "longName": "--string",
+  "required": false,
   "shortName": "-s",
   "value": undefined,
+  "values": undefined,
 }
 `;
 
-exports[`CommandLineParameter parses an input with NO parameters 6`] = `
+exports[`CommandLineParameter parses an input with NO parameters 9`] = `
+Object {
+  "argumentName": "TEXT",
+  "defaultValue": "123",
+  "description": "A string with a default",
+  "environmentVariable": "ENV_INTEGER",
+  "kind": 3,
+  "longName": "--string-with-default",
+  "required": false,
+  "shortName": undefined,
+  "value": "123",
+  "values": undefined,
+}
+`;
+
+exports[`CommandLineParameter parses an input with NO parameters 10`] = `
 Object {
   "argumentName": "LIST",
+  "defaultValue": undefined,
   "description": "A string list",
+  "environmentVariable": "ENV_INTEGER",
   "kind": 4,
   "longName": "--string-list",
+  "required": false,
   "shortName": "-l",
+  "value": undefined,
   "values": Array [],
 }
 `;
 
 exports[`CommandLineParameter prints the action help 1`] = `
-"usage: example do-job [-h] [--choice-with-default {one,two}] [-c {one,two}]
-                      [-f] [-i NUMBER] [-s TEXT] [-l LIST]
+"usage: example do-job [-h] [-c {one,two,three,default}]
+                      [--choice-with-default {one,two,three,default}] [-f]
+                      [-i NUMBER] [--integer-with-default NUMBER]
+                      --integer-required NUMBER [-s TEXT]
+                      [--string-with-default TEXT] [-l LIST]
                       
 
 a longer description
 
 Optional arguments:
   -h, --help            Show this help message and exit.
-  --choice-with-default {one,two}
+  -c {one,two,three,default}, --choice {one,two,three,default}
+                        A choice
+  --choice-with-default {one,two,three,default}
                         A choice with a default
-  -c {one,two}, --choice {one,two}
-                        A choice without a default
   -f, --flag            A flag
   -i NUMBER, --integer NUMBER
                         An integer
+  --integer-with-default NUMBER
+                        An integer with a default
```

Comment:
It'd be useful if the tool gave the environment variable and the default value.

Reply:
Good idea -- fixed in iteration 6

---
In reply to: [184545551](https://github.com/Microsoft/web-build-tools/pull/624#discussion_r184545551) [](ancestors = 184545551)